### PR TITLE
feat(terraform): Add new checks for ensuring execution history logging and Xray for State Machine is enabled 

### PR DESCRIPTION
**Pull request by reemodo**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/terraform/checks/resource/aws/StateMachineLoggingExecutionHistory.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from checkov.terraform.checks.resource.base_resource_value_check import BaseResourceValueCheck
+from checkov.common.models.enums import CheckCategories
+
+
+class StateMachineLoggingExecutionHistory(BaseResourceValueCheck):
+
+    def __init__(self):
+        name = "Ensure State Machine has X-Ray tracing enabled"
```
Comment (ChanochShayner) [inline checkov/terraform/checks/resource/aws/StateMachineLoggingExecutionHistory.py:8]: Please fix the name

File: checkov/terraform/checks/resource/aws/StateMachineXray.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+from checkov.common.models.enums import CheckCategories
+
+
+class StateMachineXray(BaseResourceValueCheck):
+
+    def __init__(self):
+        name = "Ensure State Machine has X-Ray tracing enabled"
+        id = "CKV_AWS_284"
+        supported_resources = ['aws_sfn_state_machine']
+        categories = [CheckCategories.LOGGING]
```
Comment (ChanochShayner) [inline checkov/terraform/checks/resource/aws/StateMachineXray.py:11]: I don't think this is logging

**Review by ChanochShayner — APPROVED** [review]
GJ 👑

File: checkov/terraform/checks/resource/aws/StateMachineXray.py (hunk 2 above)
Comment (gruebel) [reply checkov/terraform/checks/resource/aws/StateMachineXray.py:11]: not perfect, but the most appropriate one, we also categories the other X-ray related checks as `LOGGING`

File: tests/terraform/checks/resource/aws/test_StateMachineLoggingExecutionHistory.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import unittest
+from pathlib import Path
+
+from checkov.runner_filter import RunnerFilter
+from checkov.terraform.checks.resource.aws.StateMachineLoggingExecutionHistory import check
+from checkov.terraform.runner import Runner
+
+
+class TestStateMachineXray(unittest.TestCase):
```
Comment (gruebel) [inline tests/terraform/checks/resource/aws/test_StateMachineLoggingExecutionHistory.py:9]:
```suggestion
class TestStateMachineLoggingExecutionHistory(unittest.TestCase):
```
🙂

**Review by gruebel — APPROVED** [review]
nice work 🥇

**Pull request merged**